nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,900 @@
+        
+        return torch.stack(mention_hidden_states)
+
+    @typecheck()
+    def forward(self, input_ids, attention_mask, token_type_ids, mention_mask, mention_loss_mask):
+        """
+        No special modification required for Lightning, define it as you normally would
+        in the `nn.Module` in vanilla PyTorch.
+        """
+        if self._cfg.tokenizer.get('library', '') == 'megatron':
+            hidden_states, _ = self.bert_model(input_ids, attention_mask, tokentype_ids=token_type_ids, lm_labels=None)
+        else:
+            hidden_states = self.bert_model(
+                input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+            )
+
+        from os.path import exists
+        global UNIQUE_EPOCH_COUNTER
+
+        # print("START DUMP hidden_states")
+        # path_to_file="/home/lilee/pickle/hidden_states"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(hidden_states, f, pickle.HIGHEST_PROTOCOL)        
+        # print("END DUMP hidden_states")
+
+
+        # intent_logits, slot_logits = self.classifier(hidden_states=hidden_states)
+        intent_logits, bio_slot_logits = self.classifier(hidden_states=hidden_states)
+
+        # print("START DUMP bio_slot_logits")
+        # path_to_file="/home/lilee/pickle/bio_slot_logits"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(bio_slot_logits, f, pickle.HIGHEST_PROTOCOL)
+        # print("END DUMP bio_slot_logits")
+
+        # current:bug; hidden_states#(batch*128*768) zzz
+        
+        # mention_hidden_states_pad: b*128*hiddenstate_dim
+        mention_hidden_states_pad = IntentBIOTypeClassificationModel.get_entity_embedding_from_hidden_states(mention_mask, hidden_states)
+        
+        """
+        RUN TIME error cause here (may because too many for loop, may have better way to implement):
+            try to implement 3.7 entity description score from ReFinED paper
+            input is:
+                mention_hidden_states_pad: b*7*hiddenstate_dim  (mention's embedding)
+                    eg. [[1, 1, 1], [1, 1, 1], [1, 1, 1], [0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0]]
+                mention_loss_mask: b*7 (mention's mask)
+                    eg. [[1, 1, 1, 0, 0, 0, 0]]
+                self.slot_descriptions: every slot class's description
+                self.get_description_embedding: a method to get one description's embedding from BERT Model
+        """
+        
+        score_matrix = [] # this is the matrix store all dot score between mention and every description embedding
```

Comment:
Also, I think we should abstract this into a separate method so that we know what exactly we want to do here (description, input and output)

Reply:
This can also make it testable